pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -180,30 +157,7 @@ func (b *HybridBotanist) DeployETCD() error {
 		}
 
 		if err := b.ApplyChartSeed(filepath.Join(chartPathControlPlane, "etcd"), b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role), nil, etcd); err != nil {
-			if role == common.EtcdRoleMain {
-				// Since we have to update volumeClaimTemplate in existing statefulset, which is forbidden
-				// by k8s. So, we have to explicitly delete the old statefulset and create new one.
-				// TODO: This is backward compatibility code and should be removed in further releases.
-				if apierrors.IsInvalid(err) {
-					if err := b.K8sSeedClient.DeleteStatefulSet(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)); err != nil && !apierrors.IsNotFound(err) {
-						return err
-					}
-
-					ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
-					defer cancel()
-					if err := b.Botanist.WaitUntilEtcdStatefulsetDeleted(ctx, role); err != nil {
-						return err
-					}
-
-					if err := b.ApplyChartSeed(filepath.Join(chartPathControlPlane, "etcd"), b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role), nil, etcd); err != nil {
-						return err
-					}
-				} else {
-					return err
-				}
-			} else {
-				return err
-			}
+			return err
 		}
 		if err := b.K8sSeedClient.DeleteService(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Can you remove this "service deletion" as well? It's no longer needed (it's here due migration in past releases).

Reply:
Done